feat: IAM identity mapping management (#22)
mumoshu/terraform-provider-eksctl: README.md

    ng2 = false,
  }
```

```console
> kubectl get no
NAME                                      STATUS                     ROLES    AGE    VERSION
ip-10-0-4-28.us-east-2.compute.internal   Ready,SchedulingDisabled   <none>   4d1h   v1.16.13-eks-ec92d4
ip-10-0-5-72.us-east-2.compute.internal   Ready                      <none>   4d1h   v1.16.13-eks-ec92d4
```

## Add aws-auth ConfigMap

> You can use `iam_identity_mapping` to grant additional AWS users or roles to operate eks cluster with  aws-auth ConfigMap. 

Reviewer: ```suggestion
You can use `iam_identity_mapping` to grant additional AWS users or roles to operate the EKS cluster by letting the provider to update the `aws-auth` ConfigMap.

To get started, add one or more `iam_identity_mapping` block(s) like in the below example:
```

```HCL
provider "eksctl" {}

locals {
  iams = [
    {
      iamarn   = "arn:aws:iam::123456789012:role/master-eks-role"
      username = "master-eks-role"
      groups = [
        "system:masters"
      ]
    },
    {
      iamarn   = "arn:aws:iam::123456789012:user/user-admin"
      username = "user-admin"
      groups = [
        "system:masters"
      ]
    },
  ]
}

resource "eksctl_cluster" "myeks" {
  name   = "myeks"
  region = "us-east-1"
  spec   = <<-EOS
  iam:
    withOIDC: true
    serviceAccounts: []
  nodeGroups:
    - name: ng1
      instanceType: t2.small
      desiredCapacity: 1
    - name: ng2
      instanceType: t2.small
      desiredCapacity: 1
  EOS

  dynamic "iam_identity_mapping" {
    for_each = local.iams
    content {
      iamarn   = iam_identity_mapping.value["iamarn"]
      username = iam_identity_mapping.value["username"]
      groups   = iam_identity_mapping.value["groups"]
    }
  }
}

output aws_auth {
  value = eksctl_cluster.myeks.aws_auth_configmap
}
```

> 

Reviewer: ```suggestion
On each `terraform apply`, the provider compares the current `aws-auth` configmap against the desired configmap contents, and run `eksctl create iamidentitymapping` to create additional mappings and `eksctl delete iamidentitymapping` to delete redundant mappings.

You can confirm the result by running `eksctl get iamidentitymapping`:
```

    content {
      iamarn   = iam_identity_mapping.value["iamarn"]
      username = iam_identity_mapping.value["username"]
      groups   = iam_identity_mapping.value["groups"]
    }
  }
}

output aws_auth {
  value = eksctl_cluster.myeks.aws_auth_configmap
}
```
On each `terraform apply`, the provider compares the current `aws-auth` configmap against the desired configmap contents, and run `eksctl create iamidentitymapping` to create additional mappings and `eksctl delete iamidentitymapping` to delete redundant mappings.

You can confirm the result by running `eksctl get iamidentitymapping`:
```console
$ eksctl get iamidentitymapping -c myeks -o yaml
- groups:
  - system:bootstrappers
  - system:nodes
  rolearn: arn:aws:iam::123456789012:role/eksctl-myeks-nodegroup-ng1-NodeInstanceRole-14SXZWF9IGX6O
  username: system:node:{{EC2PrivateDNSName}}
- groups:
  - system:bootstrappers
  - system:nodes
  rolearn: arn:aws:iam::123456789012:role/eksctl-myeks-nodegroup-ng2-NodeInstanceRole-2IGYK2W51ZHJ
  username: system:node:{{EC2PrivateDNSName}}
- groups:
  - system:masters
  rolearn: arn:aws:iam::123456789012:role/admin-role
  username: admin-role
- groups:
  - system:masters

>   rolearn: arn:aws:iam::123456789012:user/user-admin

Reviewer 2: ```suggestion
  userarn: arn:aws:iam::123456789012:user/user-admin
```

>   rolearn: arn:aws:iam::123456789012:user/user-admin

Author: ```suggestion
  userarn: arn:aws:iam::123456789012:user/user-admin
```
